wordpress-mobile/WordPress-Android
# Prepublishing Nudges : Disable Publish Date on Private Posts.

Fixes #11889

## Solution
1. The Publish Date Label is bind to the `PostStatus` so that it knows it should be `Immediately` once the post is a `PRIVATE` one. 

2. When we are changing the `PostStatus` to `PRIVATE` we always overwrite the date to be sure that is is `Now` or the `CurrentDate` so the `Immediately` label is returned. 

3. The `UiState` powering the home menu actions were given an `enabled` flag so the enabled/disabled state could be reflecting in the color of the text and the `OnClickListener` enabled state could be toggled.

## Testing

1. Create a new post. 
2. Click Publish. 
3. Change the `Visibility` to `Private`. 
4. Observe that the publish date menu action is grey and it can't be clicked and it says `Immediately`. 

Screenshot | Functionality In Action 
--------|-------
   <kbd><img src="https://user-images.githubusercontent.com/1509205/82775732-283c6980-9e0e-11ea-9aa4-23e57e2dfd66.png" width="320"></kbd>  |       <kbd><img src="https://user-images.githubusercontent.com/1509205/82775740-2f637780-9e0e-11ea-83a0-74d6602efaf7.gif" width="320"></kbd>

------------

1. Create a new post. 
2. Click Publish.
3. Change the `Publish Date` to one in the future. 
4. Observe that the date is seen in the Publish Date label and it can be clicked.
3. Now change the `Visibility` to `Private`. 
4. Observe that the publish date menu action is grey and it can't be clicked and it says `Immediately`. 

Before | After 
--------|-------
   <kbd><img src="https://user-images.githubusercontent.com/1509205/82775871-9c770d00-9e0e-11ea-8f4b-0f70631bfbba.png" width="320"></kbd>  |       <kbd><img src="https://user-images.githubusercontent.com/1509205/82775886-a993fc00-9e0e-11ea-881a-5c69bc397a30.gif" width="320"></kbd>


## Reviewing 
1. Only 1 reviewer is needed but anyone can review. 
2. Ensure that https://github.com/wordpress-mobile/WordPress-Android/pull/11889 is merged. 
3. Change the base branch to `feature/prepublishing_nudges` 
4. Remove the `Not Ready For Merge` label. 
5. Merge :) 


## Submitter Checklist
